acceptance/tests/direct_puppet/cached_catalog_remediate_local_drift.rb

```diff
@@ -51,41 +51,69 @@
 
       step "Add test file resource to site.pp on master with agent-specific file path" do
         apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
-        File {
-          owner => #{master.puppet['user']},
-          group => #{master.puppet['group']},
-        }
-
-        file { "site.pp":
-          ensure => file,
-          path => "#{basedir}/environments/production/manifests/site.pp",
-          content => "node default { file { 'foo_file': ensure => file, path => '#{agent_test_file_path}', source => 'puppet:///modules/foo/foo.txt' } }",
-          mode => "0640",
-        }
-
-        file { "foo_file":
-          ensure => file,
-          path => "#{module_dir}/foo/files/foo.txt",
-          content => "code_version_1",
-          mode => "0640",
-        }
+File {
+  owner => #{master.puppet['user']},
+  group => #{master.puppet['group']},
+}
+
+file { "#{basedir}/environments/production/manifests/site.pp" :
+  ensure => file,
+  mode => "0640",
+  content => "node default {
+  file { '#{agent_test_file_path}' :
+    ensure => file,
+    source => 'puppet:///modules/foo/foo.txt'
+  }
+}",
+}
+
+file { "#{module_dir}/foo/files/foo.txt" :
+  ensure => file,
+  content => "code_version_1",
+  mode => "0640",
+}
 MANIFEST
       end
 
       step "agent: #{agent}: Initial run: create the file with code version 1 and cache the catalog"
       on(agent, puppet("agent", "-t", "--server #{master}"), :acceptable_exit_codes => [0,2])
 
+      # When there is no drift, there should be no request made to the server
+      # for file metadata or file content.  A puppet run depending on
+      # a non-server will fail if such a request is made.  Verify the agent
+      # sends a report.
+
+      step "Remove existing reports from server reports directory"
+      on(master, "rm -rf /opt/puppetlabs/server/data/puppetserver/reports/#{agent.hostname}/*")
+      r = on(master, "ls /opt/puppetlabs/server/data/puppetserver/reports/#{agent.hostname} | wc -l").stdout.chomp
+      assert_equal(r, '0', "reports directory should be empty!")
+
+      step "Verify puppet run without drift does not make file request from server"
+      r = on(agent, puppet("agent",
+        "--use_cached_catalog",
+        "--server", "no_such_host",
+        "--report_server", master.hostname,
+        "--onetime",
+        "--no-daemonize",
+        "--detailed-exitcodes",
+        "--verbose"
+      )).stderr
+      assert_equal(r, "", "Fail: Did agent try to contact server?")
+
+      step "Verify report was delivered to server"
+      r = on(master, "ls /opt/puppetlabs/server/data/puppetserver/reports/#{agent.hostname} | wc -l").stdout.chomp
```

Comment:
The path on disk will be based on the agent's node name. By default, that will be the agent's fqdn (from the perspective of the agent), which may be different from the name that the beaker uses to ssh to the agent (which is `agent.hostname`). When using the vmpooler the agent's `fqdn` and `agent.hostname` are the same, but there are cases where they may not be.

In this test, I think we want to use `agent.node_name` which will ask the node what it thinks its name is, and should always match what ends up on disk. /cc @johnduarte